corehq/util/metrics/__init__.py

```diff
@@ -0,0 +1,43 @@
+from typing import Iterable
```

Comment:
Once this is all finalized, it'd be helpful to include a module level docstring here or in `docs/` explaining how to interact with this module, perhaps with some guidance on when to use which sort of metric.